liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -966,6 +967,49 @@ public void checkConfigFile(File configFile, File serverDir) {
             }
         }
 
+        @Override
+        public boolean serverFeaturesModified(File configDirectory) {
+            ServerFeatureUtil servUtil = getServerFeatureUtil();
+            servUtil.setSuppressLogs(true); // suppress logs from ServerFeatureUtil, otherwise will flood dev console
+            // get server features from the config directory, exclude generated-features.xml
+            Set<String> generatedFiles = new HashSet<String>();
+            generatedFiles.add(BinaryScannerUtil.GENERATED_FEATURES_FILE_NAME);
+            Set<String> features = servUtil.getServerFeatures(configDirectory, serverXmlFile,
+                    new HashMap<String, File>(), generatedFiles);
+            // exclude generated features from the features list
+            Set<String> generatedFeatures = servUtil.getServerXmlFeatures(null,
+            new File(configDirectory, BinaryScannerUtil.GENERATED_FEATURES_FILE_PATH), null, null);
+            servUtil.setSuppressLogs(false); // re-enable logs from ServerFeatureUtil
+
+            if (features != null) {
+                // remove generated features in the case that they have been manually added to server config
```

Comment:
I'm wondering if we should not remove the generated features from the `features` list. I think the expectation would be that if a user adds a previously generated feature to the server config, we will remove it from our generated file.

Reply:
Removing the generated features from the features list here enables us to catch the case you are describing, where we would still want to re-generate so that those features are removed from the generated-features.xml file.